js/ui/scheduler/ui.scheduler.appointment_model.js

```diff
@@ -492,10 +492,12 @@ class AppointmentModel {
     appointmentTakesSeveralDays(appointment) {
         const dataAccessors = this._dataAccessors;
         const startDate = dataAccessors.getter.startDate(appointment);
-        const endDate = dataAccessors.getter.endDate(appointment);
+        const endDate = new Date(dataAccessors.getter.endDate(appointment));
```

Comment:
new Date() wrapping is needed in this case, since the end date may be a timestamp. Also the method **trimTime** expects in its arguments exactly the date object, not the timestamp